### PR TITLE
Image add crossOrigin settings (#143)

### LINKED ISSUE #143: SecurityError: Failed to execute 'toDataURL' on 'HTMLCanvasElement': Tainted canvases may not be exported.
I trying to download the QR code by click a button. And i get error:
`SecurityError: Failed to execute 'toDataURL' on 'HTMLCanvasElement': Tainted canvases may not be exported.`
This is my code for action click button download QR code.

```
const downloadQR = () => {
  const canvas = document.getElementById("123456");
  const pngUrl = canvas
    .toDataURL("image/png")
    .replace("image/png", "image/octet-stream");
  let downloadLink = document.createElement("a");
  downloadLink.href = pngUrl;
  downloadLink.download = "123456.png";
  document.body.appendChild(downloadLink);
  downloadLink.click();
  document.body.removeChild(downloadLink);
};
```
I found some solution is add `crossOrigin="anonymous"` to include image in QR code, refer [here](https://stackoverflow.com/questions/20424279/canvas-todataurl-securityerror).
Can u check it in your code from this [line](https://github.com/zpao/qrcode.react/blob/6aeb42abc26ffecc868b630b6ad8f507d2125813/src/index.js#L347)?

### PR DESCRIPTION
**Pull request by catmiao8**

### CODE REVIEW

**Comment by zpao** [comment]
Please describe what your change is achieving / what problem this solves

**Comment by zpao** [comment]
Oh I see this is for #143. Looks like you just made the changes in SVG, not Canvas

**Comment by catmiao8** [comment]
> 哦，我看到这是#143的。看起来您只是在 SVG 中进行了更改，而不是在 Canvas 中进行了更改

sorry  I

**Comment by catmiao8** [comment]
> Oh I see this is for #143. Looks like you just made the changes in SVG, not Canvas

Sorry, I forgot to change the canvas. It has been modified now.

**Comment by catmiao8** [comment]
Hello, are there any questions about this PR?

File: src/index.tsx [hunk 1]
```diff
@@ -344,6 +348,7 @@ const QRCodeCanvas = React.forwardRef(function QRCodeCanvas(
           setIsImageLoaded(true);
         }}
         ref={_image}
+        crossOrigin={imageSettings?.crossOrigin ?? 'anonymous'}
```
Comment (zpao) [inline src/index.tsx]: I don't _love_ that this is duplicating the work in `getImageSettings`. I'd need to see if we can pull the call to that out of our `useEffect` in here or maybe just pull the default setting out to ensure no potential drift.

File: src/index.tsx [hunk 2]
```diff
@@ -26,6 +26,7 @@ type ImageSettings = {
   x?: number;
   y?: number;
   opacity?: number;
+  crossOrigin?: 'anonymous' | 'use-credentials' | '';
```
Comment (zpao) [inline src/index.tsx]: This is absolutely correct but using union types can make using the library harder (it's why `level` is a string). I'll need to see what we can do about that.

Comment (zpao) [reply src/index.tsx]: Landed #291 which did that and it's fine, so carry on.

**Comment by zpao** [comment]
I was playing around some more and unfortunately hit the one easy case where this is a problem: localhost out to the web (at least on my webserver which is used in the example code to fetch my favicon). Need to think about it a bit more. Defaulting to `anonymous` is probably the best behavior but it would break some set of use cases. As is we would have no way to enable the broken use cases since it's only by omission of the property that it works, so would probably want to end up supporting `null` as a value with explicit intent, leaving `undefined` to hit the default value path.

**Comment by kiner-tang** [comment]
The same issue in antd: https://github.com/ant-design/ant-design/issues/48759.
When will we release a version to support this?

**Pull request merged**